rpc/src/v1/helpers/dispatch.rs

```diff
@@ -677,9 +677,7 @@ pub fn execute<D: Dispatcher + 'static>(
 			if accounts.is_hardware_address(&address) {
 				let signature = accounts.sign_message_with_hardware(&address, &data)
 					.map(|s| H520(s.into_electrum()))
-					.map(RpcH520::from)
 					.map(ConfirmationResponse::Signature)
-					// TODO: is this correct? I guess the `token` is the wallet in this context
```

Comment:
Out-of-scope from this PR but this comment was introduced by me for a long time ago and should be removed!